fission/src/systems/preferences/PreferencesSystem.ts

```diff
@@ -0,0 +1,125 @@
+import { DefaultFieldPreferences, DefaultGlobalPreferences, DefaultRobotPreferences, FieldPreferences, FieldPreferencesKey, GlobalPreference, RobotPreferences, RobotPreferencesKey } from "./PreferenceTypes"
+
+class PreferencesSystem {
+    private static _preferences: { [key: string]: Object }
+    private static _localStorageKey = "Preferences"
+
+    /** Sets a global preference to be a value of a specific type */
+    public static setGlobalPreference<T extends Object>(key: GlobalPreference, preference: T) {
+        if (this._preferences == undefined) this.loadPreferences()
+
+        this._preferences[key] = preference
+    }
```

Comment:
Could you setup some custom events for when preferences change. I would add custom event type to wrap some relevant data in, then create functions inside of preference manager to handle the boiler plate of subscribing to an event. Use the `window.addEventListener` and `window.dispatchEvent` for events. *Those may not be the correct ones, but nearly. See my poker branch for how the UI knows when the mouse is clicked*